### ISSUE TITLE
Missing normalization

### ISSUE THREAD

**Issue by ksakmann** [deeplearning-wisc/ink#1]
Hi, interesting work!
I am wondering about the normalization of the vectors. I see the prototypes $\mu_i$ are unit normalized as in your paper, but I can't see the normalization for the $z$ used to compute `pred` .  They should both live on the hypersphere for the vMF distribution, but the features come right out of the `intermediate_forward` , I am referring to this part of the code: 
```
    def postprocess(self, net: nn.Module, data: Any):
        features = net.intermediate_forward(data)
        logits = torch.matmul(features, torch.transpose(self.prototypes, 0, 1)) / self.temperature
        energy = self.temperature * torch.logsumexp(logits, dim=1)
        pred = torch.argmax(logits, dim=1)
        return pred, energy
``` 
https://github.com/deeplearning-wisc/ink/blob/0709e0c20094df81fb15266355bf787b82559409/openood/postprocessors/ink_postprocessor.py#L48

At least for  resnet50 the intermediate_forward is not normalized
https://github.com/deeplearning-wisc/ink/blob/0709e0c20094df81fb15266355bf787b82559409/openood/networks/resnet50.py#L47

Is the normalization maybe missing (or am I missing something here)?

**Comment by Top34051** [comment]
Hi, thank you for your interest in our work! In the OpenOOD framework, you could use the `CIDERNet` network which is a wrapper over your ResNets and will do the normalization when you perform the `intermediate_forward` call.

https://github.com/deeplearning-wisc/ink/blob/main/openood/networks/cider_net.py#L33

**Comment by ksakmann** [comment]
Thanks, now I get it. I was only looking for `resnet50` in the code because I saw that in the paper.  `cider_net` does provide normalized intermediate features.
